lib/packwerk/commands/help_command.rb

```diff
@@ -0,0 +1,25 @@
+# typed: strict
+# frozen_string_literal: true
+
+module Packwerk
+  module Commands
+    class HelpCommand < BaseCommand
+      extend T::Sig
+
+      sig { override.returns(Cli::Result) }
+      def run
+        Cli::Result.new(status: true, print_as_error: true, message: <<~USAGE)
+          Usage: #{$PROGRAM_NAME} <subcommand>
+
+          Subcommands:
+            init - set up packwerk
+            check - run all checks
+            update-todo - update package_todo.yml files
+            validate - verify integrity of packwerk and package configuration
+            version - output packwerk version
+            help  - display help information about packwerk
```

Comment:
Maybe we could use the registry pattern just for help messages? So have some registry that commands can call to register their help message, but let the actual command class be autoloaded.

Reply:
> registry pattern

Yup! See my other comment with a link to a specific commit. That's what I used first. But that would mean we need to eager load CLI commands.

Reply:
Not necessarily. There could be a registry that can be pre-populated for in-gem commands. For commands added externally, the process would be to register your extension (for the help message) and also create an autoloadable constant, which would not be loaded until the command was actually used.

Reply:
Ahh, so instead of registering the commands from inside the command class, we could do it outside (in a place that we know gets loaded on init), like so?

```rb
Commands.register("update-todo", help: "update package_todo.yml files", aliases: ["update"])
```

For the gem's core commands, these would all be located in the same location, while plugin gems would have to do that somewhere, where they get loaded (eg. their own `lib/packwerk/plugin_name.rb` file).

`#register` could possibly also offer a way to have command and class name be different, like so:

```rb
Commands.register("something", help: "does something", classname: "MyCommandClass")
```

Reply:
Yeah right exactly 👍

Reply:
Some autoload tests caught me loading more than I should:
<img width="968" alt="image" src="https://user-images.githubusercontent.com/816901/235637069-326facf2-7a54-46c1-8199-74960f9a180e.png">

(Didn't know about `bin/rake test:loading`.)

So I added one more commit: 721b197 (Move command registration to Cli in order to avoid early loading)